src/main/java/com/dataloom/neuron/audit/AuditEntitySet.java

```diff
@@ -0,0 +1,179 @@
+ * system-wide dependencies
+ */
+public class AuditEntitySet {
+
+    private static final Logger logger = LoggerFactory.getLogger( AuditEntitySet.class );
+
+    private static final String AUDIT_ENTITY_SET_NAME = "Loom Audit Entity Set";
+    private static final String LOOM_AUDIT_NAMESPACE  = "LOOM_AUDIT";
+
+    private static final FullQualifiedName DETAILS_PT_FQN = new FullQualifiedName( LOOM_AUDIT_NAMESPACE, "DETAILS" );
+    private static final FullQualifiedName TYPE_PT_FQN    = new FullQualifiedName( LOOM_AUDIT_NAMESPACE, "TYPE" );
+    private static final FullQualifiedName AUDIT_ET_FQN   = new FullQualifiedName( LOOM_AUDIT_NAMESPACE, "AUDIT" );
+
+    // TODO: where does this belong?
+    public static final Principal LOOM_PRINCIPAL = new Principal( PrincipalType.USER, "Loom" );
+
+    private static Collection<PropertyType> PROPERTIES;
+    private static PropertyType             TYPE_PROPERTY_TYPE;
+    private static PropertyType             DETAILS_PROPERTY_TYPE;
+    private static EntityType               AUDIT_ENTITY_TYPE;
+    private static EntitySet                AUDIT_ENTITY_SET;
+
+    private final DatasourceManager dataSourceManager;
+    private final EdmManager        entityDataModelManager;
+
+    public AuditEntitySet( DatasourceManager dataSourceManager, EdmManager entityDataModelManager ) {
+
+        this.dataSourceManager = dataSourceManager;
+        this.entityDataModelManager = entityDataModelManager;
+
+        if ( entityDataModelManager.checkEntitySetExists( AUDIT_ENTITY_SET_NAME ) ) {
+            initialize();
+        } else {
+            createAuditEntitySet();
+        }
+    }
+
+    private void initialize() {
+
+        TYPE_PROPERTY_TYPE = entityDataModelManager.getPropertyType( TYPE_PT_FQN );
+        DETAILS_PROPERTY_TYPE = entityDataModelManager.getPropertyType( DETAILS_PT_FQN );
+        AUDIT_ENTITY_TYPE = entityDataModelManager.getEntityType( AUDIT_ET_FQN );
+        AUDIT_ENTITY_SET = entityDataModelManager.getEntitySet( AUDIT_ENTITY_SET_NAME );
+
+        setProperties();
+    }
+
+    private void createAuditEntitySet() {
+
+        TYPE_PROPERTY_TYPE = new PropertyType(
+                TYPE_PT_FQN,
+                "Type",
+                Optional.of( "The type of event being logged." ),
+                ImmutableSet.of(),
+                EdmPrimitiveTypeKind.String
+        );
+
+        DETAILS_PROPERTY_TYPE = new PropertyType(
+                DETAILS_PT_FQN,
+                "Details",
+                Optional.of( "Any details about the event being logged." ),
+                ImmutableSet.of(),
+                EdmPrimitiveTypeKind.String
+        );
+
+        AUDIT_ENTITY_TYPE = new EntityType(
+                AUDIT_ET_FQN,
+                "Loom Audit",
+                "The Loom Audit Entity Type.",
+                ImmutableSet.of(),
+                Sets.newLinkedHashSet( Sets.newHashSet( TYPE_PROPERTY_TYPE.getId() ) ),
```

Comment:
May want to guard against the case where some of the properties/entity type are already created, but entity set is not. In that case, the property type Id's/entity type Id's should be obtained by creating property type/creating entity type first, check if they return null and do a getPropertyType or getEntityType in that case.